apps/admin/frontend/src/components/save_backend_file_modal.tsx

```diff
@@ -0,0 +1,271 @@
+import React, { useContext, useState } from 'react';
+import styled from 'styled-components';
+import { basename, join } from 'path';
+import {
+  assert,
+  throwIllegalValue,
+  assertDefined,
+  Result,
+  sleep,
+} from '@votingworks/basics';
+import {
+  isElectionManagerAuth,
+  isSystemAdministratorAuth,
+  usbstick,
+} from '@votingworks/utils';
+import type { ExportDataError } from '@votingworks/admin-backend';
+
+import {
+  Button,
+  Modal,
+  H1,
+  P,
+  useExternalStateChangeListener,
+  Font,
+} from '@votingworks/ui';
+
+import { AppContext } from '../contexts/app_context';
+import { Loading } from './loading';
+
+export const UsbImage = styled.img`
+  margin-right: auto;
+  margin-left: auto;
+  height: 200px;
+`;
+
+interface SaveAsButtonProps {
+  onSave: (location: string) => void;
+  options?: KioskBrowser.SaveDialogOptions;
+}
+
+export function SaveAsButton({
```

Comment:
Is this intended to be used outside of this file? If so, can we break it out to a separate component file?

Reply:
The intention is, for now, no. I thought about the possibility of using it later, but it could be broken out when that time comes.

Reply:
b00d3b2c0